src/Microsoft.AspNetCore.Routing/Tree/UrlMatchingNode.cs

```diff
@@ -18,6 +18,8 @@ public UrlMatchingNode(int length)
 
         public int Length { get; }
 
+        public bool IsCatchAll { get; set; }
```

Comment:
Its like 'am i a catchall node'. the catchall child is captured by `CatchAlls` property.